stock_request/models/stock_request_order.py

```diff
@@ -244,11 +247,17 @@ def action_confirm(self):
         return True
 
     def action_draft(self):
-        self.mapped("stock_request_ids").action_draft()
+        if self.stock_request_ids:
+            self.mapped("stock_request_ids").action_draft()
+        else:
+            self.state = "draft"
         return True
 
     def action_cancel(self):
-        self.mapped("stock_request_ids").action_cancel()
+        if self.stock_request_ids:
```

Comment:
By definition, an empty request will never be cancel.

Reply:
> Create an empty stock request order.

Don't allow to save an empty request order then.

Reply:
Ok, another use case.

- Create a stock request order with several stock_request_ids and save.
- Remove `stock_request_ids` from the stock request order. 
- It is not possible to cancel the empty stock request order.

Also in this case should we not allow to save empty stock request order? The process in that case should be just to cancel it, right?

Reply:
2 options:

- Auto-remove empty request orders when they are "emptied".
- Allow to remove empty request orders by any user authorized to add them.